core/v2/assignment.go

```diff
@@ -2,72 +2,191 @@ package v2
 
 import (
 	"fmt"
-	"math"
-	"math/big"
+	"sort"
 
-	"github.com/Layr-Labs/eigenda/common"
 	"github.com/Layr-Labs/eigenda/core"
 )
 
-// getMaxStakePercentage calculates the maximum stake percentage across specified quorums for a specific operator.
-// The function assumes that it will be passed an operator state that contains the quorums for which the operator is a member.
-// Therefore, it does not return an error if a quorum is not found in the state.
-func getMaxStakePercentage(state *core.OperatorState, quorums []core.QuorumID, id core.OperatorID) (*big.Float, error) {
-	maxStakePercentage := new(big.Float)
-	found := false
+func getOrderedOperators(state *core.OperatorState, quorum core.QuorumID) ([]core.OperatorID, map[core.OperatorID]*core.OperatorInfo, error) {
 
-	for _, q := range quorums {
-		total, ok := state.Totals[q]
-		if !ok {
+	if state == nil {
+		return nil, nil, fmt.Errorf("state cannot be nil")
+	}
+
+	operators, ok := state.Operators[quorum]
+	if !ok || len(operators) == 0 {
+		return nil, nil, fmt.Errorf("no operators found for quorum %d", quorum)
+	}
+
+	orderedOps := make([]core.OperatorID, 0, len(operators))
+	for id := range operators {
+		orderedOps = append(orderedOps, id)
+	}
+
+	sort.Slice(orderedOps, func(i, j int) bool {
+		return orderedOps[i].Hex() < orderedOps[j].Hex()
+	})
+
+	return orderedOps, operators, nil
+}
+
+func GetAssignmentsForQuorum(state *core.OperatorState, blobParams *core.BlobVersionParameters, quorum core.QuorumID) (map[core.OperatorID]*Assignment, error) {
```

Comment:
I think adding a few lines of comments on each of the major methods in this class would be helpful. From a code review perspective, it would be nice to be able to match the intended behavior of each function against its implementation.